### PR TITLE
[Bugfix] Sticky height when anchored + Position when anchors are outs…

### PR DESCRIPTION
**Pull request by gehasia**
Fixes https://github.com/zurb/foundation-sites/issues/8435
Fixes bad position of a sticky element when it's container is in another div than his top anchor.
Fixes anchored element not following the flow when resizing the window and going over/behind other elements (see https://github.com/zurb/foundation-sites/issues/8587 )
I made a video showing this :
[Reflow bug](http://sendvid.com/52a7xxmg)

### CODE REVIEW

File: js/foundation.sticky.js [hunk 1]
```diff
@@ -215,9 +215,6 @@ class Sticky {
                   * @event Sticky#stuckto
                   */
                  .trigger(`sticky.zf.stuckto:${stickTo}`);
-    this.$element.on("transitionend webkitTransitionEnd oTransitionEnd otransitionend MSTransitionEnd", function() {
-      _this._setSizes();
-    });
```
Comment (kball) [inline js/foundation.sticky.js]: We are we removing _this._setSizes after transition ends?

**Comment by kball** [comment]
@colin-marshall you've been digging around with Sticky, does this change make sense?

**Comment by gehasia** [comment]
@kball We are talking about various issues here :
https://github.com/zurb/foundation-sites/issues/8645

The first patch you merged was not good, i made a better patch since then, but in the same time @colin-marshall was working on another bugfix which was close ( https://github.com/zurb/foundation-sites/issues/8435 ) 
His solution was based on a data-static-height option, and has some bugs remainings. 
I made another fix (wich is this PR), which fix the same thing than the colin experiment, but without the need of the data-static-height option, and fixing other bugs in the same time.
I will redo a PR on the fluid height transition (the first patch, the one you marked in this comment) after all the sticky bugfixing stuff merged. 
But yes, this code is not valid anymore, waiting for transitionEnd make some glitchy artifacts.

File: js/foundation.sticky.js (hunk 1 above)
Comment (colin-marshall) [inline js/foundation.sticky.js]: @gehasia is removing this technically a part of the fix for #8435 or is it just being removed in the same commit? Can we please separate these into two different commits if they're not dependent on each other? It will make tracking progress easier as we tackle Sticky's bugs. Thanks!

**Comment by colin-marshall** [comment]
@gehasia I think you have found the fix for #8435! I tried the `bottom: auto` but the offset was off and for some reason I didn't think of just fixing it in the JS. Nice work!

Please see my diff comment upthread about separating the issues out to different commits. I'm going to get us organized tomorrow with a "master" issue that lists all outstanding sticky issues with their progress. That should make the road ahead clearer for all of us. Thanks!

File: js/foundation.sticky.js (hunk 1 above)
Comment (gehasia) [inline js/foundation.sticky.js]: It was just removed in the same commit, it's not linked. I will separate it, and include those lines deletion when i'll do the correct PR for fluid height transition (which will replace this 3 lines)

**Comment by gehasia** [comment]
Hi i'll work on it tomorrow. I'll submit clean PR for each bugs.

**Comment by gehasia** [comment]
Ok i've separated it into different PR :
https://github.com/zurb/foundation-sites/pull/8713
https://github.com/zurb/foundation-sites/pull/8714
https://github.com/zurb/foundation-sites/pull/8715

There is also a Doc + Test PR which need to be applied in order to test correctly :
https://github.com/zurb/foundation-sites/pull/8674

I close this one.
